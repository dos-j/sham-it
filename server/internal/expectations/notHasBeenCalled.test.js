const notHasBeenCalled = require("./notHasBeenCalled");

describe("unit: notHasBeenCalled", () => {
  let handler;
  let requestStore;

  beforeEach(() => {
    requestStore = [];
    handler = notHasBeenCalled(requestStore);
  });

  test("it should return 204 No Content if there are no requests in the requestStore", () => {
    expect(handler()).toEqual({ status: 204 });
  });

  test("it should return 417 Expectation failed if there are requests in the requestStore", () => {
    requestStore.push({
      request: {
        method: "GET",
        pathname: "/test",
        query: { a: 1234 },
        headers: { accept: "application/json" }
      }
    });

    expect(handler()).toEqual(
      expect.objectContaining({
        status: 417
      })
    );
  });

  test("the error message should match the snapshot", () => {
    requestStore.push({
      request: {
        method: "GET",
        pathname: "/test",
        query: { a: 1234 },
        headers: { accept: "application/json" }
      }
    });
    requestStore.push({
      request: {
        method: "POST",
        pathname: "/test/1234",
        body: { a: 1234 },
        headers: { accept: "application/json" }
      }
    });
    requestStore.push({
      request: {
        method: "POST",
        pathname: "/test/abc",
        body: "test",
        headers: { accept: "application/json" }
      }
    });

    expect(handler().body).toMatchSnapshot();
  });
});
