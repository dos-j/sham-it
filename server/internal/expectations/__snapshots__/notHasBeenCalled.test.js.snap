// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`unit: notHasBeenCalled the error message should match the snapshot 1`] = `
"Expected sham not to be called but it was called with:
  GET /test?a=1234
    accept: application/json,
  POST /test/1234
    accept: application/json
    body:
        {\\"a\\":1234},
  POST /test/abc
    accept: application/json
    body:
        test"
`;
